src/pyk/proof/reachability.py

```diff
@@ -33,12 +34,21 @@ class APRProof(Proof):
     """
 
     kcfg: KCFG
+    dependencies: list[KClaim]
     logs: dict[str, tuple[LogEntry, ...]]
 
-    def __init__(self, id: str, kcfg: KCFG, logs: dict[str, tuple[LogEntry, ...]], proof_dir: Path | None = None):
+    def __init__(
+        self,
+        id: str,
+        kcfg: KCFG,
+        logs: dict[str, tuple[LogEntry, ...]],
+        proof_dir: Path | None = None,
+        dependencies: Iterable[KClaim] = (),
+    ):
```

Comment:
```suggestion
    def __init__(
        self,
        id: str,
        kcfg: KCFG,
        logs: dict[str, tuple[LogEntry, ...]],
        proof_dir: Path | None = None,
        dependencies: Iterable[APRProof] = (),
    ):
```

So we are storing subproofs directly.